runtime/complex.go

```diff
@@ -130,3 +131,40 @@ func complexCoerce(o *Object) (complex128, bool) {
 	}
 	return complex(floatO, 0.0), true
 }
+
+func complexArithmeticOp(f *Frame, method string, v, w *Object, fun func(v, w *Object) complex128) (*Object, *BaseException) {
+	return NewComplex(fun(v, w)).ToObject(), nil
+}
+
+func complexAdd(f *Frame, v, w *Object) (*Object, *BaseException) {
+	if w.isInstance(ComplexType) {
+		return complexArithmeticOp(f, "__add__", v, w, complexAddComplex)
+	}
+
+	if w.isInstance(IntType) {
+		return complexArithmeticOp(f, "__add__", v, w, complexAddInt)
+	}
+
+	if w.isInstance(FloatType) {
+		return complexArithmeticOp(f, "__add__", v, w, complexAddFloat)
+	}
+	return NotImplemented, nil
+}
+
+func complexAddComplex(v, w *Object) complex128 {
+	lhs := toComplexUnsafe(v).Value()
+	rhs := toComplexUnsafe(w).Value()
+	return complex(real(lhs)+real(rhs), imag(lhs)+imag(rhs))
```

Comment:
Go supports complex addition, you can just do: `return lhs + rhs`

Because this is so simple and it's only used in one place you can just do this operation in line and remove the complexAddComplex function.